src/labkey/Domain.ts

```diff
@@ -406,6 +407,42 @@ export function save(config: SaveDomainOptions): XMLHttpRequest {
     });
 }
 
+export interface UpdateDomainParams {
+    createFields?: []
+    deleteFields?: [number]
+    domainId: number
+    includeWarnings?: boolean
+    updateFields?: []
+}
+
+export interface UpdateDomainOptions extends UpdateDomainParams, RequestCallbackOptions {
+    containerPath?: string
+}
+
+export function updateDomain(config: UpdateDomainOptions): XMLHttpRequest {
```

Comment:
For these primary function definitions (e.g. `updateDomain`, `listDomain`, `getPropertyUsages`, etc) I would recommend adding at least a blurb of documentation that describes intent/usage.